### ISSUE TITLE
Citations get parsed within inline source code environment

### ISSUE THREAD

**Issue by Fiech** [riboseinc/asciidoctor-bibliography#91]
Citations within the asciidoctor inline source code environment (backticks) get parsed. This may not be desired as source code environments are often used to actually explain source code behaviour.

So `` `cite:[some_key]` `` gets parsed to `(Autor, Year)` or whatever style you're using, instead of just staying `cite:[some_key]`.

**Comment by paolobrasolin** [comment]
Thanks @Fiech. Good point!

This make sense to me, as it would be impossible to write documentation about `asciidoctor-bibliography` using `asciidoctor-bibliography`. cc @ronaldtse 

While I'm at it I could protect all contexts where citations should not be checked (especially since comments already are). I can think of:

- [X] ~inline comments~
- [X] ~block comments~
- [ ] inline code
- [ ] block code
- [ ] passthrough
- [ ] ...

Basically, we want to protect contexts where a crossref would not work in vanilla `asciidoc`.

**Comment by ronaldtse** [comment]
Agree!
